### PR TITLE
Add support for Gradle 8.0

### PR DESCRIPTION
**Pull request by dwalluck**
For NCL-7660

### CODE REVIEW

File: fix-build.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+             -e 's|archiveClassifier\.set(\(.*\))|classifier = \1|g;' \
+             -e 's|archiveClassifier = |classifier = |g;' \
+             -e 's|${archiveClassifier}|${classifier}|g;' \
+             -e 's|archiveFileName = |archiveName |g;' \
+             -e 's|destinationDirectory = |destinationDir |g;' \
+             -e 's|archiveBaseName = |baseName = |g;' \
+             -e 's|${archiveBaseName}|${baseName}|g;'
+        ;;
+    *)
+        echo "Not modifiying build"
```
Comment (rnc) [inline fix-build.sh:19]:
```suggestion
        echo "Not modifying build"
```

Also, I was wondering about putting this under the gradle directory to keep the top level 'neat' ?

Comment (dwalluck) [reply fix-build.sh]: Done.

**Review by rnc — APPROVED** [review]

**Comment by rnc** [comment]
Great work!

File: build.gradle.kts [hunk 2]
```diff
@@ -331,9 +339,7 @@ subprojects {
 
         tasks.withType<ShadowJar>().configureEach {
             // ensure that a single jar is built which is the shadowed one
-            // Using non-deprecated archiveClassifier.set doesn't seem to work.
-            @Suppress("DEPRECATION")
-            classifier = ""
+            archiveClassifier.set("")
```
Comment (dwalluck) [inline build.gradle.kts:342]: I think this works now. It could be that we updated the shadow plugin version?

**Pull request merged**